platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1069,12 +1069,13 @@ - (id)mgl_jsonExpressionObject {
         }
             
         case NSKeyPathExpressionType: {
-            NSArray *expressionObject;
-            for (NSString *pathComponent in self.keyPath.pathComponents.reverseObjectEnumerator) {
+            NSMutableArray *expressionObject;
+            NSArray *keyPath = [self.keyPath componentsSeparatedByString:@"."];
```

Comment:
Ooh, good catch – I was probably thinking of `NSString.stringByDeletingPathExtension` here.